### DISCUSSION TITLE
[Feature] Improve library Folder Structure for Improved Image Organization

### DISCUSSION THREAD

**Discussion by DaanHend** [immich-app/immich#5648] [Feature Request]
### The feature

Hi,

Currently I use Synology photos as my main production photo storage. Immich is used for testing purposes to maybe move to it on a later stage. 
I have looked through some feature requests. But havent found one similar. 

Synology photos uses a upload folder structure where it separates uploads based on the device followed by the local albums of the device iteself. 
![Structure](https://github.com/immich-app/immich/assets/62287900/4b05b0a1-5486-4701-8826-5e2df7c5f814)

It would be really nice to be able to have this structure inside the library folder. Because the original images are now just one big pile and not very friendly. As well for the timeline, but that's another feature I think. 

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
You can customize the folder structure in the admin settings. For preserving the albums from a mobile device, see #1678

**Reply by DaanHend** [reply]
Yes. I knew that. Those are albums created in Immich itself.  
Immich is a backup tool first as I read multiple times. So sorting from the first insertion of images should be not a nice to have. 

Its already done perfect with dates etc. But thats only datetimes. The other options are depending on Immich itself. Its still a big messy pile if you receive and create all kinds of photos over different devices over different apps.

**Reply by bo0tzz** [reply]
You can use {{album}} in the storage template. What are you still missing exactly?

**Reply by DaanHend** [reply]
That would be different. Its based on the album within an existing folder structure based on folder naming. What I mean is that the device folders (or albums if you call them like that) themselves are structured into the upload folder. Like the image I sent earlier. 

So Device name (For if you upload from multiple devices), Application type (Whatsapp creates 5 folders or so / RAW / DCIM / etc. ). 


Would also be nice to have a way to seperate it in UI, because I stopped backing up whatsapp, Because its just a mess with all of them in one timeline.

**Reply by DaanHend** [reply]
Hope that makes it more clear
